src/builder.rs

```diff
@@ -607,123 +597,85 @@ pub mod mock {
         /// builds and returns mintnodes, spentbooks, genesis_dbc_shares, and genesis dbc
         #[allow(clippy::type_complexity)]
         pub fn build(
-            self,
-            rng8_initial: &mut (impl rand8::RngCore + rand_core::CryptoRng + Clone),
+            mut self,
+            rng8: &mut (impl rand8::RngCore + rand_core::CryptoRng),
         ) -> Result<(
             Vec<MintNode<SimpleKeyManager>>,
             Vec<SpentBookNodeMock>,
-            Vec<GenesisDbcShare>,
             Dbc,
+            GenesisMaterial,
+            AmountSecrets,
         )> {
-            let mut spent_proof_shares: Vec<SpentProofShare> = Default::default();
-            let mut genesis_set: Vec<GenesisDbcShare> = Default::default();
             let mut mint_nodes: Vec<MintNode<SimpleKeyManager>> = Default::default();
-            let mut spentbook_nodes: Vec<SpentBookNodeMock> = Default::default();
 
-            for mint_node in self.mint_nodes.into_iter() {
-                // Each MintNode must start from same random seed when issuing Genesis.
-                let mut rng8 = rng8_initial.clone();
+            // note: rng is necessary for RingCtMaterial::sign().
```

Comment:
I skimmed the blst-ringct / blst-bulletproofs crates to see how they use `rng` in signing, I think it's safe to say that we don't really need deterministic values here, the rng is used in the rangeproofs and in the pseudo-commitements.

So, even though it would be nice to have a deterministic `sign()` method, it shouldn't matter which RNG is used in practice to build the genesis as everyone can unblind the values and see that the genesis was build using the correct amounts.